packages/core/src/machines/actors/auth/signIn.ts

```diff
@@ -0,0 +1,280 @@
+                },
+              ],
+              onError: [
+                {
+                  cond: 'shouldRedirectToConfirmSignUp',
+                  target: '#signInActor.rejected',
+                },
+                {
+                  actions: 'setRemoteError',
+                  target: 'edit',
+                },
+              ],
+            },
+          },
+        },
+      },
+      confirmSignIn: {
+        initial: 'edit',
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              SIGN_IN: '#signInActor.signIn',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          submit: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'confirmSignIn',
+              onDone: {
+                target: '#signInActor.resolved',
+                actions: ['setUser', 'clearChallengeName'],
+              },
+              onError: {
+                target: 'edit',
+                actions: 'setRemoteError',
+              },
+            },
+          },
+        },
+      },
+      forceNewPassword: {
+        initial: 'edit',
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              SIGN_IN: '#signInActor.signIn',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          submit: {
+            entry: 'clearError',
+            invoke: {
+              src: 'forceNewPassword',
+              onDone: {
+                actions: ['setUser', 'clearChallengeName'],
+                target: '#signInActor.resolved',
+              },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      setupTOTP: {
+        initial: 'edit',
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              SIGN_IN: '#signInActor.signIn',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          submit: {
+            entry: [sendUpdate(), 'clearError'],
+            invoke: {
+              src: 'verifyTotpToken',
+              onDone: {
+                actions: ['setUser', 'clearChallengeName'],
+                target: '#signInActor.resolved',
+              },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      resolved: {
+        type: 'final',
+        data: (context) => ({
+          user: context.user,
+        }),
+      },
+      rejected: {
+        type: 'final',
+        data: (context) => ({
+          intent: 'confirmSignUp',
+          authAttributes: {
+            username: context.formValues.username,
```

Comment:
If we get rejected due to unconfirmed user, we send back data with intent that we should `confirmSignUp`.